### PR TITLE
Create Secure Parser for Unmarshalling SP File Content

### PR DESCRIPTION
**Pull request by hisanhunais**
### Proposed changes in this pull request
- Creating a secure parser for Unmarshalling Service Provider File Content

### Approach
1. Introduced a secure SAXParserFactory
2. Parsed the contained through the SAXParserFactor and generated a source object
3. Perform unmarshall operation by passing the generated source object to the unmarshaller

### Related PRs
https://github.com/wso2-support/carbon-identity-framework/pull/1550

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 1]
```diff
@@ -1981,12 +1984,28 @@ private ServiceProvider unmarshalSP(SpFileContent spFileContent, String tenantDo
...
+                spf.setFeature(Constants.SAX_FEATURE_PREFIX + Constants.EXTERNAL_GENERAL_ENTITIES_FEATURE, false);
+                spf.setFeature(Constants.SAX_FEATURE_PREFIX + Constants.EXTERNAL_PARAMETER_ENTITIES_FEATURE, false);
+                spf.setFeature(Constants.XERCES_FEATURE_PREFIX + Constants.LOAD_EXTERNAL_DTD_FEATURE, false);
+                spf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            } catch (SAXException | ParserConfigurationException e) {
+                log.error("Failed to load XML Processor Feature " + Constants.EXTERNAL_GENERAL_ENTITIES_FEATURE + " or "
+                        + Constants.EXTERNAL_PARAMETER_ENTITIES_FEATURE + " or " + Constants.LOAD_EXTERNAL_DTD_FEATURE
+                        + " or secure-processing.");
+            }
+            //Creating source object using the secure parser
```
Comment (ashensw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2001]:
```suggestion
            // Creating source object using the secure parser.
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 2]
```diff
@@ -1981,12 +1984,28 @@ private ServiceProvider unmarshalSP(SpFileContent spFileContent, String tenantDo
...
+            Source xmlSource = new SAXSource(spf.newSAXParser().getXMLReader(),
+                    new InputSource(new StringReader(spFileContent.getContent())));
+            //Performing unmarshall operation by passing the generated source object to the unmarshaller
```
Comment (ashensw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2004]:
```suggestion
            // Performing unmarshall operation by passing the generated source object to the unmarshaller.
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java (hunk 1 above)
Comment (hisanhunais) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/3472/commits/9dcda46c7661222562f4f62ed300680446725b2f

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java (hunk 2 above)
Comment (hisanhunais) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/3472/commits/9dcda46c7661222562f4f62ed300680446725b2f

**Review by ashensw — APPROVED** [review]

**Pull request merged**